native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/utils/ClientSampleSyntaxTreeUtils.java

```diff
@@ -328,49 +327,65 @@ private static VariableDeclarationNode getInputDeclarationStatement(Method metho
 
     private static ArrayList<Node> getFieldNodes(Message message, Map<String, Message> msgMap) {
         ArrayList<Node> nodes = new ArrayList<>();
+        if (message.getFieldList().size() == 0) {
+            return nodes;
+        }
         for (Field field : message.getFieldList()) {
             nodes.add(NodeFactory.createFieldMatchPatternNode(
                     AbstractNodeFactory.createIdentifierToken(
                             field.getFieldName() + " "), SyntaxTreeConstants.SYNTAX_TREE_COLON,
-                    getFieldPatternNode(field, msgMap, false)));
+                    getFieldPatternNode(field, msgMap)));
             nodes.add(NodeFactory.createCaptureBindingPatternNode(SyntaxTreeConstants.SYNTAX_TREE_COMMA));
         }
         nodes.remove(nodes.size() - 1);
         return nodes;
     }
 
-    private static Node getFieldPatternNode(Field field, Map<String, Message> msgMap, boolean isRepeated) {
+    private static Node getFieldPatternNode(Field field, Map<String, Message> msgMap) {
+        Node node;
         switch (field.getFieldType()) {
             case "int":
             case "float":
             case "decimal":
-                return getCaptureBindingPatternNode("1");
+                node = getCaptureBindingPatternNode("1");
```

Comment:
If we can maintain a constant map with all these values with the field type as a key. I can avoid this switch case. it is an improvement.

Reply:
Tracked in this https://github.com/ballerina-platform/ballerina-standard-library/issues/3731